src/status_im2/subs/wallet_2/wallet.cljs

```diff
@@ -0,0 +1,40 @@
+(ns status-im2.subs.wallet-2.wallet
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [utils.number]))
+
+(defn- calculate-raw-balance
+  [raw-balance decimals]
+  (if-let [n (utils.number/parse-int raw-balance nil)]
+    (/ n (Math/pow 10 (utils.number/parse-int decimals)))
+    0))
+
+(defn- total-per-token
+  [item]
+  (reduce (fn [ac balances]
+            (+ (calculate-raw-balance (:rawBalance balances)
+                                      (:decimals item))
+               ac))
+          0
+          (vals (:balancesPerChain item))))
```

Comment:
This refactor is really good, If I remember correctly there were two nested `reduce` calls.
Extracting it into a function is clean.